modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperienceServiceImpl.java

```diff
@@ -131,6 +131,19 @@ public SegmentsExperience fetchSegmentsExperience(
 		return segmentsExperience;
 	}
 
+	@Override
+	public long getDraftLayoutPlid(long classPK) {
+		Layout layout = _layoutLocalService.fetchLayout(classPK);
+
+		Layout draftLayout = layout.fetchDraftLayout();
+
+		if (draftLayout != null) {
+			return draftLayout.getPlid();
+		}
+
+		return classPK;
+	}
+
```

Comment:
I don't think this should go here. Segments experiences are independent of layouts (even if now only layouts are supported).

Why not add this as a private method in the display context?

Reply:
The main reason was because I didn't want to add another service (util) to the display context and I though that the segment experience could be the responsible for that